synedrion/src/paillier/encryption.rs

```diff
@@ -294,7 +294,9 @@ impl<P: PaillierParams> Ciphertext<P> {
         // To isolate `rho`, calculate `(rho^N)^(N^(-1)) mod N`.
         // The order of `Z_N` is `phi(N)`, so the inversion in the exponent is modulo `phi(N)`.
         let sk_inv_modulus = sk.inv_modulus();
-        let randomizer_mod = Secret::init_with(|| ciphertext_mod_n.pow(sk_inv_modulus));
+        let randomizer_mod = Secret::init_with(|| {
+            ciphertext_mod_n.pow_bounded_exp(sk_inv_modulus.expose_secret(), sk_inv_modulus.bound())
```

Comment:
Shouldn't `pow()` work here?

Reply:
It should, good catch. This was a left-over from early experimentation.